test/io/test_tim_grabber.cpp

```diff
@@ -0,0 +1,106 @@
+#include <pcl/test/gtest.h>
+#include <pcl/point_types.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/io/tim_grabber.h>
+
+#include <string>
+#include <random>
+
+using PointT = pcl::PointXYZ;
+using CloudT = pcl::PointCloud<PointT>;
+
+class TestableTimGrabber : public pcl::TimGrabber {
+  using TimGrabber::TimGrabber;
+  public:
+		using TimGrabber::point_cloud_xyz_ptr_;
+		using TimGrabber::processTimPacket;
+		using TimGrabber::updateLookupTables;
+		using TimGrabber::toPointClouds;
+};
+
+std::vector<std::string> packets_;
+std::vector<CloudT> correct_clouds_;
+TestableTimGrabber grabber_;
+
+class TimGrabberTest : public ::testing::Test {
+  protected:
+    TimGrabberTest () = default;
+    ~TimGrabberTest () override {}
+
+    static void SetUpTestCase () {
+    }
+
+    static void TearDownTestCase () {
+    }
+
+    virtual void SetUp () {
```

Comment:
```suggestion
    virtual void SetUp() override {
```

Reply:
some more places missing the override keyword.

Reply:
Thanks.

may I use `SetUp () override` instead of `SetUp() override`?

And removed unused functions!

Reply:
I don't mind, but I think if you run the make format (not entirely sure how that part works :P ) the extra parentheses will get removed anyway. Noticed it when reviewing another PR which just does that (correcting format).